connect/runtime/src/main/java/org/apache/kafka/connect/runtime/HerderConnectorContext.java

```diff
@@ -38,4 +38,9 @@ public void requestTaskReconfiguration() {
         // Distributed herder will forward the request to the leader if needed
         herder.requestTaskReconfiguration(connectorName);
     }
+
+    @Override
+    public void raiseError(Exception e) {
+        herder.onFailure(connectorName, e);
```

Comment:
Sorry, I take that back. It just requires digging deep enough to find the synchronization...